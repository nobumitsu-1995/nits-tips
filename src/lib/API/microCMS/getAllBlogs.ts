import type { MicroCMS } from '@/types/microCMS'
import { client } from './client'

type Blog = MicroCMS['blog']

const ENDPOINT = 'blogs'

export const getAllBlogs = async () => {
  const data = await client.getAllContents<Blog>({
    endpoint: ENDPOINT,
    queries: {
      filters: `isHidden[not_equals]true`,
    },
  })

  if (!data) return []

  return data
}
