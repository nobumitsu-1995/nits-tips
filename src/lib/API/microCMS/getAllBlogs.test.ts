import { getAllBlogs } from './getAllBlogs'

const mockedJson = vitest.fn().mockResolvedValue({})
const mockedFetch = vitest.fn()

beforeEach(() => {
  const mockResponse = {
    ok: true,
    status: 200,
    json: mockedJson,
  }
  global.fetch = mockedFetch.mockResolvedValue(mockResponse)
})

afterEach(() => {
  vitest.resetAllMocks()
})

test('getAllBlogsで正常に値が取得できる', async () => {
  await getAllBlogs()
  expect(mockedFetch).toHaveBeenCalledWith(
    expect.stringContaining(
      'blogs?filters=isHidden%5Bnot_equals%5Dtrue&limit=0',
    ),
    expect.any(Object),
  )
  expect(mockedJson).toHaveBeenCalledWith()
})
