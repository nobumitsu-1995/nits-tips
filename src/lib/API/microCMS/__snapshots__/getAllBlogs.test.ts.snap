// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getAllBlogsで正常に値が取得できる 1`] = `
{
  "json": [MockFunction spy] {
    "calls": [
      [],
    ],
    "results": [
      {
        "type": "return",
        "value": Promise {},
      },
    ],
  },
  "ok": true,
  "status": 200,
}
`;
